src/Platforms/AbstractPlatform.php

```diff
@@ -2912,12 +2912,33 @@ public function getColumnDeclarationSQL($name, array $column)
      */
     public function getDecimalTypeDeclarationSQL(array $column)
     {
-        $column['precision'] = ! isset($column['precision']) || empty($column['precision'])
-            ? 10 : $column['precision'];
-        $column['scale']     = ! isset($column['scale']) || empty($column['scale'])
-            ? 0 : $column['scale'];
+        if (! empty($column['precision'])) {
+            $precision = $column['precision'];
+        } else {
+            Deprecation::trigger(
+                'doctrine/dbal',
+                'https://github.com/doctrine/dbal/pull/5637',
+                'Relying on the default decimal column precision is deprecated'
+                    . ', specify the precision explicitly.',
+            );
+
+            $precision = 10;
+        }
+
+        if (isset($column['scale'])) {
```

Comment:
Because scale can be empty (specifically, `0`) which shouldn't trigger a deprecation notice. But now I realize that by just checking the scale for `isset()` we can break the code that relies on the `empty()` check and passes NULL.

This code can be improved.